src/sknnr/_gnn_rda.py

```diff
@@ -0,0 +1,48 @@
+from sklearn.base import TransformerMixin
+
+from ._base import TransformedKNeighborsRegressor, YFitMixin
+from .transformers import RDATransformer
+
+
+class GNNRDARegressor(YFitMixin, TransformedKNeighborsRegressor):
```

Comment:
Reading through your post more carefully, I think this is the same idea you were proposing for combining `CCATransformer` and `RDATransformer` into `ConstrainedTransformer`, right?

>  We may want to think about providing the method (i.e. "cca" vs. "rda" vs. others yet to be implemented) as a hyperparameter to a ConstrainedTransformer.

In both cases, having one parameterized class vs. two classes both seem like valid approaches to me, and I think we could end up with similarly clean code either way if we used inheritance to reduce duplication with the latter option. I'm happy to leave the decision up to if you have any preference either way!

Reply:
> Reading through your post more carefully, I think this is the same idea you were considering for combining `CCATransformer` and `RDATransformer` into `ConstrainedTransformer`, right?

Yes, this was my initial thought and it's the way I currently have it implemented.  `CCATransformer` and `RDATransformer` have now been replaced with just `ConstrainedTransformer` which now takes a `method` keyword argument (currently "cca" or "rda" and defaulting to "cca").  There's a bit of logic to test for and initialize the correct `ConstrainedOrdination` subclass, but that's really the only change between what was `CCATransformer` and `ConstrainedTransformer`.

Likewise, `GNNRegressor` now has the same `method` keyword argument and internally its transformer is just a `ConstrainedTransformer` instance (rather than a `CCATransformer` instance).

Does this seem like a valid approach?

Probably, the most substantive change I made was to place `ConstrainedOrdination`, `CCA`, and `RDA` all into their own separate files.  I'm not really sure this is necessary and am fine if they all live in the same file (probably `transformers/_constrained_ordination.py`).

Reply:
Nice, that refactor cleaned up the duplication a lot! I think as long as the difference between `CCA` and `RDA` feels minor enough to you for it to be a parameter, that's probably the right call. The downside is having to validate the user string, but I think it's probably worth the reduced complexity versus using inheritance.

Regarding the file organization, I have a slight preference for combining them into a single file, but could be persuaded if you disagree.

Reply:
> Regarding the file organization, I have a slight preference for combining them into a single file, but could be persuaded if you disagree.

Yes, I was leaning that way myself and will make this change.

> I think as long as the difference between CCA and RDA feels minor enough to you for it to be a parameter, that's probably the right call. The downside is having to validate the user string, but I think it's probably worth the reduced complexity versus using inheritance.

Actually, I'm now a bit curious how you would implement this via inheritance?  I'm not seeing that pattern.  If it's not too much trouble to sketch it out and you'd prefer that design, I'm happy to consider it.

Reply:
I think this could be done with inheritance by keeping a `ConstrainedTransformer` class like you currently have (minus the `method` parameter) and having `CCATransformer` and `RDATransformer` subclasses that only override an abstract `fit` with their respective ordinations. 

But I don't have any preference for that over the solution you went with, especially if we're going to end up with more constrained methods.

Reply:
Sorry to drag out this discussion, but wouldn't we essentially be "kicking the can" of changing which *transformer* to use (e.g. `CCATransformer`, `RDATransformer`) in `GNNRegressor` based on `method` versus changing which *ordination* to use (e.g. `CCA`, `RDA`) in `ConstrainedTransformer` based on `method`?  I feel like I'm not seeing something ...

Reply:
Yeah, I don't think there's any benefit to the implementation from keeping `CCATransformer` and `RDATransformer` separate. The only meaningful difference would be the public API, but I don't think that's necessarily a benefit either.